ui/browsing/artistpage.go

```diff
@@ -443,32 +441,36 @@ func NewArtistPageHeader(page *ArtistPage) *ArtistPageHeader {
 
 func (a *ArtistPageHeader) Clear() {
 	a.artistID = ""
+	a.artistImageID = ""
 	a.favoriteBtn.IsFavorited = false
 	a.titleDisp.Segments[0].(*widget.TextSegment).Text = ""
 	a.biographyDisp.Text = lang.L(artistBioNotAvailableKey)
 	for _, obj := range a.similarArtists.Objects {
 		obj.Hide()
 	}
 	a.artistImage.SetImage(nil, false)
+	a.fullSizeCoverFetching = false
 }
 
-func (a *ArtistPageHeader) Update(artist *mediaprovider.ArtistWithAlbums) {
+func (a *ArtistPageHeader) Update(artist *mediaprovider.ArtistWithAlbums, im *backend.ImageManager) {
 	if artist == nil {
 		return
 	}
 	a.favoriteBtn.IsFavorited = artist.Favorite
 	a.favoriteBtn.Refresh()
 	a.artistID = artist.ID
+	a.artistImageID = artist.CoverArtID
```

Comment:
We should add back the `if artist.CoverArtID == ""` check and only assign `a.artistImageID` if it is not empty. In the UpdateInfo function, where it falls back to the LastFM image URL if CoverArtID was not present. I think the `showPopUpCover` function will then need two codepaths - the one that you've added for when `artistImageID` is not empty, and falling back to the original codepath for the case where ImageURL was used instead (in which case the image is a fixed size anyway).

The complication here is because the Subsonic API supports both of these systems for artist art, and some servers may support only one of the two.